packages/connect/src/device/Device.ts

```diff
@@ -555,10 +555,10 @@ export class Device extends TypedEmitter<DeviceEvents> {
                         // set the timeout for this call so whenever it happens "unacquired device" will be created instead
                         // next time device should be called together with "Initialize" (calling "acquireDevice" from the UI)
                         new Promise((_resolve, reject) => {
-                            getFeaturesTimeoutId = setTimeout(
-                                () => reject(new Error('GetFeatures timeout')),
-                                getFeaturesTimeout,
-                            );
+                            getFeaturesTimeoutId = setTimeout(async () => {
+                                await cancelPrompt(this, false);
```

Comment:
maybe more defensive approach?
```suggestion
                                cancelPrompt(this, false).finally(() => reject(new Error('GetFeatures timeout')));
```